### PR TITLE
Issues #2, #7, #12, #16, #23 Combined Unit and UI Test Coverage using JaCoCo

### PR DESCRIPTION
**Pull request by Neeham**
#2, #7, #12, #16, #23 :
 
We can now generate a combined Unit and UI test coverage using JaCoCo.

How to view the UI and Unit test in one JaCoCo report?

Step 1) Open Android Studio

Step 2) Click on Terminal (it's located at the bottom left, last option)

Step 3) Enter the command to execute the report (This takes about 1m 30s, the more test we have, the longer the time gets)

Command to execute combined UI and Unit test in one  JaCoCo report:

``gradlew clean combinedTestReportDebug``

Important information: This command relies on the fact that all UI and Unit tests are passing. If one fails, it stops and will give an error on the command line as to which test failed its execution!

What does this command do? 

1) It first runs the UI test (you will see your phone magically opening the app and doing things)

2) It then runs the Unit Test

3) It finally combines the 2 previous test in one JaCoCo report.

The report is stored under (Project View): 
SOEN390-CampusGuideMap\app\build\reports\jacocoCombined\debug\index.html

### CODE REVIEW

File: app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt [hunk 1]
```diff
@@ -149,7 +149,7 @@ class CalendarFragmentTest {
         //checks if there exists a button with text "okay" within the calendar selection widget an performs a click on it
         val appCompatButton2 = onView(
             allOf(
-                withId(android.R.id.button1), withText("Okay"),
+                withId(android.R.id.button1), withText("OK"),
```
Comment (JeremiahTee) [inline app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt]: Change "OK" to "OKAY". It fixes the failure at this line for this test

File: app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt [hunk 2]
```diff
@@ -149,7 +149,7 @@ class CalendarFragmentTest {
         //checks if there exists a button with text "okay" within the calendar selection widget an performs a click on it
```
Comment (JeremiahTee) [inline app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt]: Change "okay" to "OKAY" to be emphasize case sensitivity. In the app the text is all caps

**Comment by JeremiahTee** [comment]
Commits look good. Amazing work :)
It's just the tests that are failing. Specifically, when I ran the command, at first, I had 3 failures. When I changed the "OK" to "OKAY" (as pointed out in the inline comment) the failures were reduced to 1.

This is the only failure I have after fixing the test:
image
 
image

Error message looks ambiguous. I'd let @makchamp check it out. 
*It might be related to testing the title of the alert dialog. "Select Your Calendars". If you remove the fix, you will see the line associated with checking that specific textview fail as well

File: app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt (hunk 1 above)
Comment (Neeham) [reply app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt]: @JeremiahTee On my end, this is actually OK and not OKAY. Maybe on newer PR this has been reworded so far with respect to master, it's actually called OK and passes the test.

File: app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt (hunk 2 above)
Comment (Neeham) [reply app/src/androidTest/java/com/droidhats/campuscompass/views/CalendarFragmentTest.kt]: Good idea! This has been fixed 😄

**Comment by Neeham** [comment]
> Commits look good. Amazing work :)
> It's just the tests that are failing. Specifically, when I ran the command, at first, I had 3 failures. When I changed the "OK" to "OKAY" (as pointed out in the inline comment) the failures were reduced to 1.
> 
> This is the only failure I have after fixing the test:
> image
> 
> image
> 
> Error message looks ambiguous. I'd let @makchamp check it out.
> *It might be related to testing the title of the alert dialog. "Select Your Calendars". If you remove the fix, you will see the line associated with checking that specific textview fail as well

Actually this issue was related to calendar not being shown on view after selecting the calendar. This has been fixed 😄 awaiting after your review @JeremiahTee 👍

**Pull request merged**